pym/bob/input.py

```diff
@@ -2373,6 +2373,10 @@ def __parse(self):
 
         # finally parse recipes
         if not os.path.isdir("recipes"):
+            self.__cache.close()
```

Comment:
No. That way YamlCache.close() is called twice because it is guarded by a try/finally statement in parse() above! Also, instead of cleaning up the mess left behind what about moving the test just above self.__cache.open() on line 2349? That way the file would not be created in the first place...